packages/acceptance-tests/pkg-tests-core/sources/utils/tests.ts

```diff
@@ -333,24 +333,13 @@ export const startPackageServer = ({type}: { type: keyof typeof packageServerUrl
         throw new Error(`Assertion failed: Invalid request type`);
 
       const {username} = parsedRequest;
-      const otp = request.headers[`npm-otp`];
 
       const user = validLogins[username];
       if (!user) {
         processError(response, 401, `Unauthorized`);
         return;
       }
 
-      if (user.requiresOtp && user.otp !== otp) {
```

Comment:
This logic got moved "to the top" so it gets used now for all authenticated requests. 

Please note that this changes the order of the checks and this might influence which error gets reported to the user. Previously you could respond with `401 Unauthorized` before even checking the OTP stuff. Now it's the other way around, the OTP stuff gets checked first.

Is it OK? Should this be refactored somehow?

Please also note that this (existing) logic here might contain a potential bug. I didn't check this super carefully, so I might be wrong, but I think that the login handler here doesn't check if the user token is the correct one for the given user. It seems that I could login as user B with the token belonging to user A.

Reply:
Sounds fair, but at best the logic must be activated twice. When inside the login codepath, the bearer token isn't there yet, so with your change the OTP logic wouldn't run. I refactored it further to move the OTP check in a function that can be called from both places 👍